internal/rpmmd/repository.go

```diff
@@ -375,3 +378,45 @@ func (packages PackageList) ToPackageInfos() []PackageInfo {
 
 	return results
 }
+
+func (r RepoConfig) MarshalJSON() ([]byte, error) {
+	type aliasType RepoConfig
+	type compatType struct {
+		aliasType
+
+		BaseURL string `json:"Baseurl,omitempty"`
```

Comment:
Repo config doesn't actually have any json tags... so I did this for now. Any suggestions welcome.

Reply:
I'm fairly sure `RepoConfig` never gets serialized to/from json. It's an internal struct that is filled in manually by `loadRepositoriesFromFile` which uses `repository` as the on-disk json struct.

Reply:
Ah right, I had to add this in though because the `regression-old-worker` test kept failing with an rpmmd error because it couldn't unmarshal the fields. 

Just to check, is it not getting marshalled here though?
https://github.com/osbuild/osbuild-composer/blob/da2e80f99f65bf708e0c7c3a31ee4844846287d6/internal/worker/json.go#L224

Reply:
Ah, yes it would appear so :/ The function says it is temporary, so maybe that will change. But I really have no idea, not having touched the worked code at all.

Reply:
It's probably time to drop the `compatJob` part here.  The change it was covering for was made almost a year ago (94c7fda779).  Since da2e80f99f we only care about and test workers from 3 versions back, so I think we're safe to remove it and it's not completely out of scope for this PR I think.

Reply:
Looking at it again, it seems that even in the new format we would serialise `rpmmd.RepoConfig` as part of the `rpmmd.PackageSet` in `DepsolveJob`.  That said, we've been serialising `rpmmd.RepoConfig` for ages, since the inception of the `DepsolveJob` 0f90aa9c782cace1fc88c39c7ec752bd1f12d799, so maybe it's fine?

Reply:
Because we're simplifying this PR, this technically could be done in a separate PR? Happy to add the json tags and create a PR for that, wdyt?

Reply:
Whatever is easier for you...

Reply:
Did it here :)